core/src/main/java/org/apache/accumulo/fate/Fate.java

```diff
@@ -226,11 +223,17 @@ public Fate(T environment, TStore<T> store, Function<Repo<T>,String> toLogStrFun
    * Launches the specified number of worker threads.
    */
   public void startTransactionRunners(AccumuloConfiguration conf) {
-    int numThreads = conf.getCount(Property.MANAGER_FATE_THREADPOOL_SIZE);
-    executor = ThreadPools.createExecutorService(conf, Property.MANAGER_FATE_THREADPOOL_SIZE);
-    for (int i = 0; i < numThreads; i++) {
-      executor.execute(new TransactionRunner());
-    }
+    final ThreadPoolExecutor pool = (ThreadPoolExecutor) ThreadPools.createExecutorService(conf,
+        Property.MANAGER_FATE_THREADPOOL_SIZE);
+    fatePoolWatcher = ThreadPools.createGeneralScheduledExecutorService(conf);
+    fatePoolWatcher.schedule(() -> {
+      ThreadPools.makeResizeable(pool, conf, Property.MANAGER_FATE_THREADPOOL_SIZE);
+      int remaining = 1000 - pool.getQueue().size();
+      for (int i = 0; i < remaining; i++) {
+        pool.execute(new TransactionRunner());
```

Comment:
This may throw a bunch of exceptions if the pool is shutdown.  Would be nice to avoid that.  Could wait for `fatePoolWatcher` to shutdown before calling `executor.shutdown()` in the shutdown() method below in this class.  That might avoid the exceptions here.

Alternatively could catch rejectedexecutionexception here and ignore it if the pool is shutdown.

Reply:
I addressed this in 8bed9e5.